service/common/src/main/java/org/apache/polaris/service/persistence/MetadataCacheManager.java

```diff
@@ -0,0 +1,155 @@
+          tableLikeEntity.getMetadataCacheContent(),
+          Stream.of(
+                  entityProperties.get(PolarisEntityConstants.ENTITY_BASE_LOCATION),
+                  entityProperties.get(
+                      IcebergTableLikeEntity.USER_SPECIFIED_WRITE_DATA_LOCATION_KEY),
+                  entityProperties.get(
+                      IcebergTableLikeEntity.USER_SPECIFIED_WRITE_METADATA_LOCATION_KEY))
+              .filter(Objects::nonNull)
+              .collect(Collectors.toSet()));
+    } else {
+      MetadataJson fallbackJson = MetadataJson.fromMetadata(fallback.get());
+      var cacheResult =
+          cacheTableMetadataJson(
+              tableLikeEntity,
+              fallbackJson.content(),
+              maxBytesToCache,
+              callContext,
+              metastoreManager,
+              resolvedEntityView);
+      if (!cacheResult.isSuccess()) {
+        LOGGER.debug(String.format("Failed to cache metadata for %s", tableIdentifier));
+      }
+      return fallbackJson;
+    }
+  }
+
+  /**
+   * Attempt to add table metadata to the cache
+   *
+   * @return The result of trying to cache the metadata
+   */
+  private static EntityResult cacheTableMetadataJson(
+      IcebergTableLikeEntity tableLikeEntity,
+      String metadataJson,
+      int maxBytesToCache,
+      PolarisCallContext callContext,
+      PolarisMetaStoreManager metaStoreManager,
+      PolarisResolutionManifestCatalogView resolvedEntityView) {
+    if (maxBytesToCache != FeatureConfiguration.METADATA_CACHE_MAX_BYTES_INFINITE_CACHING) {
+      if (metadataJson.length() > maxBytesToCache) {
+        LOGGER.debug(
+            String.format(
+                "Will not cache metadata for %s; metadata above the limit of %d bytes",
+                tableLikeEntity.getTableIdentifier(), maxBytesToCache));
+        return new EntityResult(EntityResult.ReturnStatus.SUCCESS, null);
+      }
+    }
+
+    LOGGER.debug(String.format("Caching metadata for %s", tableLikeEntity.getTableIdentifier()));
+    TableLikeEntity newTableLikeEntity =
+        new IcebergTableLikeEntity.Builder(tableLikeEntity)
+            .setMetadataContent(tableLikeEntity.getMetadataLocation(), metadataJson)
+            .build();
+    PolarisResolvedPathWrapper resolvedPath =
+        resolvedEntityView.getResolvedPath(
+            tableLikeEntity.getTableIdentifier(),
+            PolarisEntityType.TABLE_LIKE,
+            PolarisEntitySubType.ICEBERG_TABLE);
+    try {
+      return metaStoreManager.updateEntityPropertiesIfNotChanged(
+          callContext,
+          PolarisEntity.toCoreList(resolvedPath.getRawParentPath()),
+          newTableLikeEntity);
+    } catch (RuntimeException e) {
+      // PersistenceException (& other extension-specific exceptions) may not be in scope,
+      // but we can make a best-effort attempt to swallow it and just forego caching
+      if (e.toString().contains("PersistenceException")) {
+        LOGGER.debug(
```

Comment:
Not sure we should change this to warn? This would be something broken and not just a "could not cache" state right?